### PR TITLE
Adding websockets for new CQRS

### PR DESCRIPTION
**Pull request by acron0**

### CODE REVIEW

File: src/witan/gateway/components/server.clj [hunk 1]
```diff
@@ -20,7 +26,9 @@
                            (-> #'app
                                (wrap-components this)
                                (wrap-content-type "application/json")
-                               (wrap-log))
+                               (wrap-cors :access-control-allow-origin [#".*"]
+                                          :access-control-allow-methods [:get :post])
+                               #_(wrap-log))
```
Comment (elisehuard) [inline src/witan/gateway/components/server.clj]: intentionally commented?

File: src/witan/gateway/handler.clj [hunk 2]
```diff
@@ -1,49 +1,131 @@
...
+                         :command/key key
+                         :command/version version
+                         :command/id id
+                         :command/receipt receipt
+                         :command/received-at now}))))
+
+(defmethod handle-message
+  :default
+  [_ msg _]
+  (println "UNKNOWN MESSAGE:" (:type msg) msg))
```
Comment (elisehuard) [inline src/witan/gateway/handler.clj]: logging?

File: src/witan/gateway/handler.clj [hunk 3]
```diff
@@ -1,49 +1,131 @@
...
+(defn ws-handler [request]
+  (let [components (:witan.gateway.components.server/components request)]
+    (with-channel request channel
+      (connect! channel)
+      (on-close channel (partial disconnect! channel))
+      (on-receive channel #(try
+                             (let [msg (read-string %)
+                                   error (rs/check-message "1.0" msg)]
+                               (if-not error
+                                 (handle-message channel msg components)
```
Comment (elisehuard) [inline src/witan/gateway/handler.clj]: so we have both a message and a command channel for the websockets

Comment (acron0) [inline src/witan/gateway/handler.clj]: No, just a single channel. The `handle-message` `defmulti` dispatches on whether it's a query or a command.

File: src/witan/gateway/components/query_router.clj [hunk 4]
```diff
@@ -1,32 +1,45 @@
...
+            [clojure.data.codec.base64  :as b64]))
 
+(defn detect-namespace
+  [p]
+  (let [x (-> p first first)]
+    (if (keyword? x)
+      (keyword (namespace x))
+      (when (and (coll? x)
+                 (keyword? (first x)))
+        (keyword (namespace (first x)))))))
```
Comment (elisehuard) [inline src/witan/gateway/components/query_router.clj]: what happens when it falls through all the conditionals?

Comment (acron0) [inline src/witan/gateway/components/query_router.clj]: It returns `nil` and fails the `contains?` check (here)[https://github.com/MastodonC/witan.gateway/blob/0db897d979cbe561205d62d42300bae54f649bb8/src/witan/gateway/components/query_router.clj#L22]

**Comment by elisehuard** [comment]
so when all comments are covered lgtm :shipit:

**Pull request merged**